wallet-new/src/Cardano/Wallet/Kernel/DB/AcidState.hs

```diff
@@ -343,6 +425,43 @@ createHdWallet newRoot utxoByAccount =
         , accountUpdate      = return () -- just need to create it, no more
         }
 
+-- | Begin restoration by creating an HdWallet with the given HdRoot,
+-- starting from the 'HdAccountOutsideK' state.
+restoreHdWallet :: HdRoot
+                -> SlotId
+                -> Map HdAccountId (Utxo, [AddrWithId])
+                -> Update DB (Either HD.CreateHdRootError ())
+restoreHdWallet newRoot slotId utxoByAccount =
+    runUpdateDiscardSnapshot . zoom dbHdWallets $ do
+      HD.createHdRoot newRoot
+      updateAccounts_ $ map mkUpdate (Map.toList utxoByAccount)
+  where
+    mkUpdate :: (HdAccountId, (Utxo, [AddrWithId]))
+             -> AccountUpdate HD.CreateHdRootError ()
+    mkUpdate (accId, (utxo, addrs)) = AccountUpdate {
+          accountUpdateId    = accId
+        , accountUpdateNew   = AccountUpdateNew utxo
```

Comment:
So just an empty utxo here?

Reply:
Yeah. Or it might be nicer still to change this so that we can actually specify that we don't want to create the account in up-to-date state but rather in restoration state.